### PR TITLE
adjust sp login ui with new ui components

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azure/toolkit/eclipse/common/component/AzureRadioButton.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        super(parent, SWT.RADIO);
+        this.debouncer = new TailingDebouncer(() -> {
+            this.fireValueChangedEvent(this.value);
+            this.doValidate();
+        }, DEBOUNCE_DELAY);
+        this.addSelectionListener(this);
+    }
+    @Override
+    public Boolean getValue() {
+        return this.getSelection();
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azure/toolkit/eclipse/common/component/AzureRadioButton.java]: use `AzureTaskManager.invokeAndWait`?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azure/toolkit/eclipse/common/component/AzureRadioButton.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Nonnull
+    @Override
+    public final AzureValidationInfo doValidate() {
+        final AzureValidationInfo validationInfo = AzureFormInputControl.super.doValidate();
+        this.setValidationInfo(validationInfo);
+        return validationInfo;
+    }
+
+    public void setValidationInfo(AzureValidationInfo info) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azure/toolkit/eclipse/common/component/AzureRadioButton.java]: remove this method?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/GuidAzureTextInput.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        this.setValidator(this::doValidateValue);
+        this.setRequired(true);
+    }
+
+    public AzureValidationInfo doValidateValue() {
+        final String value = this.getValue();
+        // validate length
+        if (!GUID_PATTERN.matcher(value).matches()) {
+            return AzureValidationInfo.builder().input(this)
+                    .message("Guid must match regex:^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/GuidAzureTextInput.java]: format String with `GUID_REGEX`?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/GuidAzureTextInput.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    public GuidAzureTextInput(Composite parent) {
+        super(parent);
+        this.setValidator(this::doValidateValue);
+        this.setRequired(true);
+    }
+
+    public AzureValidationInfo doValidateValue() {
+        final String value = this.getValue();
+        // validate length
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/GuidAzureTextInput.java]: Remove comment or add logic to validate length?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java [hunk 5]
```diff
@@ -146,13 +99,14 @@ protected Control createDialogArea(Composite parent) {
         gdCompositeCert.widthHint = 191;
         compositeCert.setLayoutData(gdCompositeCert);
 
-        txtCertificate = new Text(compositeCert, SWT.BORDER | SWT.READ_ONLY);
+        txtCertificate = new AzureTextInput(compositeCert, SWT.BORDER | SWT.READ_ONLY);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java]: add `txtCertificate.setLabeledBy(lblSecret);` ?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java [hunk 6]
```diff
@@ -184,141 +140,76 @@ public void widgetSelected(SelectionEvent e) {
...
-        return container;
     }
 
-    private void uiTextComponents2Json() {
-        if (txtJson.isDisposed()) {
+    private void triggerPasswordCertState() {
+        boolean selection = radioCertificate.getSelection();
+        btnOpenFileButton.setEnabled(selection);
+        txtCertificate.setEnabled(selection);
+        txtPassword.setEnabled(!selection);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java]: add `setRequired`?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java (hunk 5 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java]: use setLabel instead

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java (hunk 6 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/login/ServicePrincipalLoginPanel.java]: no use

**Review by Flanker32 — APPROVED** [review]
LGTM

**Pull request merged**